dadguide/database_manager.py

```diff
@@ -96,15 +99,23 @@ def __init__(self, data_file):
         self._con = lite.connect(data_file, detect_types=lite.PARSE_DECLTYPES)
         self._con.row_factory = lite.Row
 
+        self.graph = None
+        self.build_graph()
 
         self.cachedmonsters = None
         self.expiry = 0
+        self.generate_all_monsters()
```

Comment:
in a subsequent pr, we should try to rip apart DadguideDatabase so that its primary function is only to query sqlite. Ideally we would change all the call sites of `get_active_skill_query`, `get_leader_skill_query`... etc to call the graph (which would have these functions) directly.